React/Base/RCTConvert.m

```diff
@@ -709,35 +819,50 @@ + (UIColor *)UIColor:(id)json
     CGFloat g = ((argb >> 8) & 0xFF) / 255.0;
     CGFloat b = (argb & 0xFF) / 255.0;
     return [UIColor colorWithRed:r green:g blue:b alpha:a];
-#if TARGET_OS_OSX // [TODO(macOS ISS#2323203)
+// [TODO(macOS ISS#2323203)
   } else if ([json isKindOfClass:[NSDictionary class]]) {
     NSDictionary *dictionary = json;
     id value = nil;
     if ((value = [dictionary objectForKey:@"semantic"])) {
       NSString *semanticName = value;
-      NSColor *color = RCTColorFromSemanticColorName(semanticName);
+      UIColor *color = RCTColorFromSemanticColorName(semanticName);
```

Comment:
Is UIColor macro'd to NSColor or something? I still think that pattern is pretty confusing. In my opinion, we should at least create an RCTUIColor that is NSColor for Mac and UIColor on iOS. Then it's at least apparent we aren't attempting to manipulate an iOS object on Mac

Reply:
Or does this just not run on Mac? I see you removed the if def, but I'm assuming we don't want to lose Mac functionality

Reply:
It is that UIColor is defined to NSColor via RCTUIKit.h.   I'll investigate refactoring stuff to RCTUIColor

Reply:
Done